purchase_sale_inter_company/models/purchase_order.py

```diff
@@ -137,14 +213,14 @@ def _prepare_sale_order_data(
         if self.notes:
             new_order.note = self.notes
         new_order.commitment_date = self.date_planned
+        if self.env.context.get("pricelist"):
+            new_order.pricelist_id = self.env.context.get("pricelist")
         return new_order._convert_to_write(new_order._cache)
 
-    def _prepare_sale_order_line_data(self, purchase_line, dest_company, sale_order):
+    def _prepare_sale_order_line_data(self, purchase_line, sale_order):
```

Comment:
Changing a method signature can be disruptive. Have you checked it usage in other modules?

Reply:
this method (I mean with such parameters) is only used here for this customer

Reply:
This is the exact change we do in OCA module to have customer's expected workflow.